2021-09-02 06:50:53,185 ERROR frappe Could not take error snapshot: 'User' object has no attribute 'docstatus'
Site: mum125.erpgulf.com
Form Dict: {'doctype': 'DocType', 'fields': '["`tabDocType`.`name`","`tabDocType`.`owner`","`tabDocType`.`creation`","`tabDocType`.`modified`","`tabDocType`.`modified_by`","`tabDocType`.`_user_tags`","`tabDocType`.`_comments`","`tabDocType`.`_assign`","`tabDocType`.`_liked_by`","`tabDocType`.`docstatus`","`tabDocType`.`parent`","`tabDocType`.`parenttype`","`tabDocType`.`parentfield`","`tabDocType`.`idx`","`tabDocType`.`module`","`tabDocType`.`color`"]', 'filters': '[["DocType","name","like","%ART%"],["DocType","name","=","Data Migration Plan Mapping"]]', 'order_by': '`tabDocType`.`modified` desc', 'start': '0', 'page_length': '20', 'view': 'List', 'with_comment_count': 'true'}
Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 54, in application
    init_request(request)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 128, in init_request
    frappe.local.http_request = frappe.auth.HTTPRequest()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 32, in __init__
    self.set_session()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 69, in set_session
    frappe.local.login_manager = LoginManager()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 122, in __init__
    self.make_session(resume=True)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 206, in make_session
    full_name=self.full_name, user_type=self.user_type)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 189, in __init__
    self.resume()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 255, in resume
    validate_ip_address(self.user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 390, in validate_ip_address
    user = frappe.get_cached_doc("User", user) if not frappe.flags.in_test else frappe.get_doc("User", user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 807, in get_cached_doc
    doc = get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 850, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 75, in get_doc
    return controller(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 113, in __init__
    self.load_from_db()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 142, in load_from_db
    if not getattr(self, "_metaclass", False) and self.meta.issingle:
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/base_document.py", line 91, in meta
    self._meta = frappe.get_meta(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 881, in get_meta
    return frappe.model.meta.get_meta(doctype, cached=cached)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 37, in get_meta
    meta = Meta(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 84, in __init__
    self.process()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 105, in process
    self.set_custom_permissions()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 420, in set_custom_permissions
    filters=dict(parent=self.name), update=dict(doctype='Custom DocPerm'))
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1433, in get_all
    return get_list(doctype, *args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1406, in get_list
    return frappe.model.db_query.DatabaseQuery(doctype).execute(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 101, in execute
    self.columns = self.get_table_columns()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 338, in get_table_columns
    return get_table_columns(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 49, in get_table_columns
    return frappe.db.get_table_columns(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/database/database.py", line 894, in get_table_columns
    raise self.TableMissingError('DocType', doctype)
pymysql.err.ProgrammingError: ('DocType', 'Custom DocPerm')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 36, in make_error_snapshot
    snapshot = get_snapshot(exception)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in get_snapshot
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/inspect.py", line 1269, in formatargvalues
    specs.append(convert(args[i]))
  File "/usr/lib64/python3.6/inspect.py", line 1266, in convert
    return formatarg(name) + formatvalue(locals[name])
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in <lambda>
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/reprlib.py", line 55, in repr
    return self.repr1(x, self.maxlevel)
  File "/usr/lib64/python3.6/pydoc.py", line 1035, in repr1
    return cram(stripid(repr(x)), self.maxother)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 1360, in __repr__
    docstatus = f" docstatus={self.docstatus}" if self.docstatus else ""
AttributeError: 'User' object has no attribute 'docstatus'
2021-09-02 06:50:53,382 ERROR frappe Could not take error snapshot: 'User' object has no attribute 'docstatus'
Site: mum125.erpgulf.com
Form Dict: {'doctype': 'DocType', 'fields': '["`tabDocType`.`name`","`tabDocType`.`owner`","`tabDocType`.`creation`","`tabDocType`.`modified`","`tabDocType`.`modified_by`","`tabDocType`.`_user_tags`","`tabDocType`.`_comments`","`tabDocType`.`_assign`","`tabDocType`.`_liked_by`","`tabDocType`.`docstatus`","`tabDocType`.`parent`","`tabDocType`.`parenttype`","`tabDocType`.`parentfield`","`tabDocType`.`idx`","`tabDocType`.`module`","`tabDocType`.`color`"]', 'filters': '[["DocType","name","=","Web Template"]]', 'order_by': '`tabDocType`.`modified` desc', 'start': '0', 'page_length': '20', 'view': 'List', 'with_comment_count': 'true'}
Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 54, in application
    init_request(request)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 128, in init_request
    frappe.local.http_request = frappe.auth.HTTPRequest()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 32, in __init__
    self.set_session()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 69, in set_session
    frappe.local.login_manager = LoginManager()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 122, in __init__
    self.make_session(resume=True)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 206, in make_session
    full_name=self.full_name, user_type=self.user_type)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 189, in __init__
    self.resume()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 255, in resume
    validate_ip_address(self.user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 390, in validate_ip_address
    user = frappe.get_cached_doc("User", user) if not frappe.flags.in_test else frappe.get_doc("User", user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 807, in get_cached_doc
    doc = get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 850, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 75, in get_doc
    return controller(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 113, in __init__
    self.load_from_db()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 142, in load_from_db
    if not getattr(self, "_metaclass", False) and self.meta.issingle:
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/base_document.py", line 91, in meta
    self._meta = frappe.get_meta(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 881, in get_meta
    return frappe.model.meta.get_meta(doctype, cached=cached)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 37, in get_meta
    meta = Meta(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 84, in __init__
    self.process()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 105, in process
    self.set_custom_permissions()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 420, in set_custom_permissions
    filters=dict(parent=self.name), update=dict(doctype='Custom DocPerm'))
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1433, in get_all
    return get_list(doctype, *args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1406, in get_list
    return frappe.model.db_query.DatabaseQuery(doctype).execute(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 101, in execute
    self.columns = self.get_table_columns()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 338, in get_table_columns
    return get_table_columns(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 49, in get_table_columns
    return frappe.db.get_table_columns(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/database/database.py", line 894, in get_table_columns
    raise self.TableMissingError('DocType', doctype)
pymysql.err.ProgrammingError: ('DocType', 'Custom DocPerm')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 36, in make_error_snapshot
    snapshot = get_snapshot(exception)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in get_snapshot
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/inspect.py", line 1269, in formatargvalues
    specs.append(convert(args[i]))
  File "/usr/lib64/python3.6/inspect.py", line 1266, in convert
    return formatarg(name) + formatvalue(locals[name])
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in <lambda>
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/reprlib.py", line 55, in repr
    return self.repr1(x, self.maxlevel)
  File "/usr/lib64/python3.6/pydoc.py", line 1035, in repr1
    return cram(stripid(repr(x)), self.maxother)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 1360, in __repr__
    docstatus = f" docstatus={self.docstatus}" if self.docstatus else ""
AttributeError: 'User' object has no attribute 'docstatus'
2021-09-02 06:50:53,593 ERROR frappe Could not take error snapshot: 'User' object has no attribute 'docstatus'
Site: mum125.erpgulf.com
Form Dict: {'doctype': 'DocType', 'fields': '["`tabDocType`.`name`","`tabDocType`.`owner`","`tabDocType`.`creation`","`tabDocType`.`modified`","`tabDocType`.`modified_by`","`tabDocType`.`_user_tags`","`tabDocType`.`_comments`","`tabDocType`.`_assign`","`tabDocType`.`_liked_by`","`tabDocType`.`docstatus`","`tabDocType`.`parent`","`tabDocType`.`parenttype`","`tabDocType`.`parentfield`","`tabDocType`.`idx`","`tabDocType`.`module`","`tabDocType`.`color`"]', 'filters': '[["DocType","name","like","%ART%"],["DocType","name","=","Onboarding Permission"]]', 'order_by': '`tabDocType`.`modified` desc', 'start': '0', 'page_length': '20', 'view': 'List', 'with_comment_count': 'true'}
Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 54, in application
    init_request(request)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 128, in init_request
    frappe.local.http_request = frappe.auth.HTTPRequest()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 32, in __init__
    self.set_session()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 69, in set_session
    frappe.local.login_manager = LoginManager()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 122, in __init__
    self.make_session(resume=True)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 206, in make_session
    full_name=self.full_name, user_type=self.user_type)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 189, in __init__
    self.resume()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 255, in resume
    validate_ip_address(self.user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 390, in validate_ip_address
    user = frappe.get_cached_doc("User", user) if not frappe.flags.in_test else frappe.get_doc("User", user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 807, in get_cached_doc
    doc = get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 850, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 75, in get_doc
    return controller(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 113, in __init__
    self.load_from_db()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 142, in load_from_db
    if not getattr(self, "_metaclass", False) and self.meta.issingle:
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/base_document.py", line 91, in meta
    self._meta = frappe.get_meta(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 881, in get_meta
    return frappe.model.meta.get_meta(doctype, cached=cached)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 37, in get_meta
    meta = Meta(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 84, in __init__
    self.process()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 105, in process
    self.set_custom_permissions()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 420, in set_custom_permissions
    filters=dict(parent=self.name), update=dict(doctype='Custom DocPerm'))
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1433, in get_all
    return get_list(doctype, *args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1406, in get_list
    return frappe.model.db_query.DatabaseQuery(doctype).execute(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 101, in execute
    self.columns = self.get_table_columns()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 338, in get_table_columns
    return get_table_columns(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 49, in get_table_columns
    return frappe.db.get_table_columns(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/database/database.py", line 894, in get_table_columns
    raise self.TableMissingError('DocType', doctype)
pymysql.err.ProgrammingError: ('DocType', 'Custom DocPerm')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 36, in make_error_snapshot
    snapshot = get_snapshot(exception)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in get_snapshot
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/inspect.py", line 1269, in formatargvalues
    specs.append(convert(args[i]))
  File "/usr/lib64/python3.6/inspect.py", line 1266, in convert
    return formatarg(name) + formatvalue(locals[name])
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in <lambda>
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/reprlib.py", line 55, in repr
    return self.repr1(x, self.maxlevel)
  File "/usr/lib64/python3.6/pydoc.py", line 1035, in repr1
    return cram(stripid(repr(x)), self.maxother)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 1360, in __repr__
    docstatus = f" docstatus={self.docstatus}" if self.docstatus else ""
AttributeError: 'User' object has no attribute 'docstatus'
2021-09-02 06:50:53,784 ERROR frappe Could not take error snapshot: 'User' object has no attribute 'docstatus'
Site: mum125.erpgulf.com
Form Dict: {'doctype': 'DocType', 'fields': '["`tabDocType`.`name`","`tabDocType`.`owner`","`tabDocType`.`creation`","`tabDocType`.`modified`","`tabDocType`.`modified_by`","`tabDocType`.`_user_tags`","`tabDocType`.`_comments`","`tabDocType`.`_assign`","`tabDocType`.`_liked_by`","`tabDocType`.`docstatus`","`tabDocType`.`parent`","`tabDocType`.`parenttype`","`tabDocType`.`parentfield`","`tabDocType`.`idx`","`tabDocType`.`module`","`tabDocType`.`color`"]', 'filters': '[["DocType","name","like","%ART%"],["DocType","name","=","Onboarding Step Map"]]', 'order_by': '`tabDocType`.`modified` desc', 'start': '0', 'page_length': '20', 'view': 'List', 'with_comment_count': 'true'}
Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 54, in application
    init_request(request)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 128, in init_request
    frappe.local.http_request = frappe.auth.HTTPRequest()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 32, in __init__
    self.set_session()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 69, in set_session
    frappe.local.login_manager = LoginManager()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 122, in __init__
    self.make_session(resume=True)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 206, in make_session
    full_name=self.full_name, user_type=self.user_type)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 189, in __init__
    self.resume()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 255, in resume
    validate_ip_address(self.user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 390, in validate_ip_address
    user = frappe.get_cached_doc("User", user) if not frappe.flags.in_test else frappe.get_doc("User", user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 807, in get_cached_doc
    doc = get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 850, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 75, in get_doc
    return controller(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 113, in __init__
    self.load_from_db()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 142, in load_from_db
    if not getattr(self, "_metaclass", False) and self.meta.issingle:
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/base_document.py", line 91, in meta
    self._meta = frappe.get_meta(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 881, in get_meta
    return frappe.model.meta.get_meta(doctype, cached=cached)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 37, in get_meta
    meta = Meta(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 84, in __init__
    self.process()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 105, in process
    self.set_custom_permissions()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 420, in set_custom_permissions
    filters=dict(parent=self.name), update=dict(doctype='Custom DocPerm'))
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1433, in get_all
    return get_list(doctype, *args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1406, in get_list
    return frappe.model.db_query.DatabaseQuery(doctype).execute(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 101, in execute
    self.columns = self.get_table_columns()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 338, in get_table_columns
    return get_table_columns(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 49, in get_table_columns
    return frappe.db.get_table_columns(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/database/database.py", line 894, in get_table_columns
    raise self.TableMissingError('DocType', doctype)
pymysql.err.ProgrammingError: ('DocType', 'Custom DocPerm')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 36, in make_error_snapshot
    snapshot = get_snapshot(exception)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in get_snapshot
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/inspect.py", line 1269, in formatargvalues
    specs.append(convert(args[i]))
  File "/usr/lib64/python3.6/inspect.py", line 1266, in convert
    return formatarg(name) + formatvalue(locals[name])
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in <lambda>
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/reprlib.py", line 55, in repr
    return self.repr1(x, self.maxlevel)
  File "/usr/lib64/python3.6/pydoc.py", line 1035, in repr1
    return cram(stripid(repr(x)), self.maxother)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 1360, in __repr__
    docstatus = f" docstatus={self.docstatus}" if self.docstatus else ""
AttributeError: 'User' object has no attribute 'docstatus'
2021-09-02 06:50:54,010 ERROR frappe Could not take error snapshot: 'User' object has no attribute 'docstatus'
Site: mum125.erpgulf.com
Form Dict: {'doctype': 'DocType', 'fields': '["`tabDocType`.`name`","`tabDocType`.`owner`","`tabDocType`.`creation`","`tabDocType`.`modified`","`tabDocType`.`modified_by`","`tabDocType`.`_user_tags`","`tabDocType`.`_comments`","`tabDocType`.`_assign`","`tabDocType`.`_liked_by`","`tabDocType`.`docstatus`","`tabDocType`.`parent`","`tabDocType`.`parenttype`","`tabDocType`.`parentfield`","`tabDocType`.`idx`","`tabDocType`.`module`","`tabDocType`.`color`"]', 'filters': '[["DocType","name","like","%ART%"],["DocType","name","=","Workspace Link"]]', 'order_by': '`tabDocType`.`modified` desc', 'start': '0', 'page_length': '20', 'view': 'List', 'with_comment_count': 'true'}
Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 54, in application
    init_request(request)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 128, in init_request
    frappe.local.http_request = frappe.auth.HTTPRequest()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 32, in __init__
    self.set_session()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 69, in set_session
    frappe.local.login_manager = LoginManager()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 122, in __init__
    self.make_session(resume=True)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 206, in make_session
    full_name=self.full_name, user_type=self.user_type)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 189, in __init__
    self.resume()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 255, in resume
    validate_ip_address(self.user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 390, in validate_ip_address
    user = frappe.get_cached_doc("User", user) if not frappe.flags.in_test else frappe.get_doc("User", user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 807, in get_cached_doc
    doc = get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 850, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 75, in get_doc
    return controller(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 113, in __init__
    self.load_from_db()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 142, in load_from_db
    if not getattr(self, "_metaclass", False) and self.meta.issingle:
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/base_document.py", line 91, in meta
    self._meta = frappe.get_meta(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 881, in get_meta
    return frappe.model.meta.get_meta(doctype, cached=cached)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 37, in get_meta
    meta = Meta(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 84, in __init__
    self.process()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 105, in process
    self.set_custom_permissions()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 420, in set_custom_permissions
    filters=dict(parent=self.name), update=dict(doctype='Custom DocPerm'))
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1433, in get_all
    return get_list(doctype, *args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1406, in get_list
    return frappe.model.db_query.DatabaseQuery(doctype).execute(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 101, in execute
    self.columns = self.get_table_columns()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 338, in get_table_columns
    return get_table_columns(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 49, in get_table_columns
    return frappe.db.get_table_columns(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/database/database.py", line 894, in get_table_columns
    raise self.TableMissingError('DocType', doctype)
pymysql.err.ProgrammingError: ('DocType', 'Custom DocPerm')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 36, in make_error_snapshot
    snapshot = get_snapshot(exception)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in get_snapshot
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/inspect.py", line 1269, in formatargvalues
    specs.append(convert(args[i]))
  File "/usr/lib64/python3.6/inspect.py", line 1266, in convert
    return formatarg(name) + formatvalue(locals[name])
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in <lambda>
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/reprlib.py", line 55, in repr
    return self.repr1(x, self.maxlevel)
  File "/usr/lib64/python3.6/pydoc.py", line 1035, in repr1
    return cram(stripid(repr(x)), self.maxother)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 1360, in __repr__
    docstatus = f" docstatus={self.docstatus}" if self.docstatus else ""
AttributeError: 'User' object has no attribute 'docstatus'
2021-09-02 06:50:54,219 ERROR frappe Could not take error snapshot: 'User' object has no attribute 'docstatus'
Site: mum125.erpgulf.com
Form Dict: {'doctype': 'DocType', 'fields': '["`tabDocType`.`name`","`tabDocType`.`owner`","`tabDocType`.`creation`","`tabDocType`.`modified`","`tabDocType`.`modified_by`","`tabDocType`.`_user_tags`","`tabDocType`.`_comments`","`tabDocType`.`_assign`","`tabDocType`.`_liked_by`","`tabDocType`.`docstatus`","`tabDocType`.`parent`","`tabDocType`.`parenttype`","`tabDocType`.`parentfield`","`tabDocType`.`idx`","`tabDocType`.`module`","`tabDocType`.`color`"]', 'filters': '[["DocType","name","like","%ART%"],["DocType","name","=","Workspace"]]', 'order_by': '`tabDocType`.`modified` desc', 'start': '0', 'page_length': '20', 'view': 'List', 'with_comment_count': 'true'}
Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 54, in application
    init_request(request)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/app.py", line 128, in init_request
    frappe.local.http_request = frappe.auth.HTTPRequest()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 32, in __init__
    self.set_session()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 69, in set_session
    frappe.local.login_manager = LoginManager()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 122, in __init__
    self.make_session(resume=True)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 206, in make_session
    full_name=self.full_name, user_type=self.user_type)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 189, in __init__
    self.resume()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/sessions.py", line 255, in resume
    validate_ip_address(self.user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/auth.py", line 390, in validate_ip_address
    user = frappe.get_cached_doc("User", user) if not frappe.flags.in_test else frappe.get_doc("User", user)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 807, in get_cached_doc
    doc = get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 850, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 75, in get_doc
    return controller(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 113, in __init__
    self.load_from_db()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 142, in load_from_db
    if not getattr(self, "_metaclass", False) and self.meta.issingle:
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/base_document.py", line 91, in meta
    self._meta = frappe.get_meta(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 881, in get_meta
    return frappe.model.meta.get_meta(doctype, cached=cached)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 37, in get_meta
    meta = Meta(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 84, in __init__
    self.process()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 105, in process
    self.set_custom_permissions()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 420, in set_custom_permissions
    filters=dict(parent=self.name), update=dict(doctype='Custom DocPerm'))
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1433, in get_all
    return get_list(doctype, *args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/__init__.py", line 1406, in get_list
    return frappe.model.db_query.DatabaseQuery(doctype).execute(*args, **kwargs)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 101, in execute
    self.columns = self.get_table_columns()
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/db_query.py", line 338, in get_table_columns
    return get_table_columns(self.doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/meta.py", line 49, in get_table_columns
    return frappe.db.get_table_columns(doctype)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/database/database.py", line 894, in get_table_columns
    raise self.TableMissingError('DocType', doctype)
pymysql.err.ProgrammingError: ('DocType', 'Custom DocPerm')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 36, in make_error_snapshot
    snapshot = get_snapshot(exception)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in get_snapshot
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/inspect.py", line 1269, in formatargvalues
    specs.append(convert(args[i]))
  File "/usr/lib64/python3.6/inspect.py", line 1266, in convert
    return formatarg(name) + formatvalue(locals[name])
  File "/opt/bench/frappe-bench/apps/frappe/frappe/utils/error.py", line 81, in <lambda>
    call = inspect.formatargvalues(args, varargs, varkw, locals, formatvalue=lambda value: '={}'.format(pydoc.text.repr(value)))
  File "/usr/lib64/python3.6/reprlib.py", line 55, in repr
    return self.repr1(x, self.maxlevel)
  File "/usr/lib64/python3.6/pydoc.py", line 1035, in repr1
    return cram(stripid(repr(x)), self.maxother)
  File "/opt/bench/frappe-bench/apps/frappe/frappe/model/document.py", line 1360, in __repr__
    docstatus = f" docstatus={self.docstatus}" if self.docstatus else ""
AttributeError: 'User' object has no attribute 'docstatus'
